### PR TITLE
State: Remove unnecessary preferences state

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

The `hasPreferencesRequestFailed` selector has been unused since #54387. When we remove it, we can also remove the `failed` preferences sub-reducer, because it's unused as well. That sub-reducer is the only consumer of the `PREFERENCES_SAVE` and `PREFERENCES_SAVE_FAILURE` actions, so we can stop dispatching those as well. Finally, we can remove some related unit tests. 

This PR removes all of the aforementioned.

#### Testing instructions

* Verify tests pass: `yarn run test-client client/state/preferences`

### CODE REVIEW

File: client/state/preferences/actions.js [hunk 1]
```diff
@@ -97,10 +90,5 @@ export const savePreference = ( key, value ) => ( dispatch ) => {
 				value,
 			} );
 		} )
-		.catch( ( error ) => {
-			dispatch( {
-				type: PREFERENCES_SAVE_FAILURE,
-				error,
-			} );
-		} );
+		.catch( () => {} );
```
Comment (flootr) [inline client/state/preferences/actions.js:93]: I wonder what should happen if that request fails. Ideally, we would show some error notice (not part of this PR I think). Right now, if I block network requests to `/me/preferences` and change the color scheme it still says _Settings saved successfully_.

**Review by flootr — APPROVED** [review]
Fine from a code perspective 👍🏻 just one note about a potential UX improvement which I think would be a good follow up (not necessarily our team).

File: client/state/preferences/actions.js (hunk 1 above)
Comment (jsnajdr) [reply client/state/preferences/actions.js:93]: For the color scheme setting specifically, the success notice [doesn't check the result](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/color-scheme-picker/index.jsx#L45) at all:
```js
saveColorSchemePreference( ... ); // let it run
successNotice( translate( 'Settings saved successfully!' ) ); // announce success immediately
```

IMO the best API is when the thunk action returns the promise unhandled, and the caller React component is responsible for handling the error.

Comment (tyxla) [reply client/state/preferences/actions.js:93]: I agree with you both, and indeed the best way is to leave the component to handle the error.

However `savePreference()` is used in plenty of places (20+ instances total) and we'd have to test all of them just for the sake of this cleanup. We had a `catch()` in the previous version of the code and it's possible one or more usages rely on the errors being caught here. 

So, I prefer keeping it as-is for now, and maybe clean up in the future if you folks are feeling strongly about it 😉

**Pull request merged**